File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/TaskBaselines/taskbaselineduration.rb
module Asposetasksjava
  module TaskBaselineDuration
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
        
        onesec = 10000000
        onemin = 60 * onesec
        onehour = 60 * onemin
        
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new
		
    		# Creating TaskBaseline:
        task = project.getRootTask().getChildren().add("Task")
        project.setBaseline(Rjb::import('com.aspose.tasks.BaselineType').Baseline)

        baseline = task.getBaselines().toList().get(0)
        duration = baseline.getDuration().toDouble()
        baseline_duration = duration / onehour
        puts baseline_duration
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithTaskBaselines/TaskBaselineDuration.php
<?php
namespace Aspose\Tasks\WorkingWithTaskBaselines;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\BaselineType as BaselineType;

class TaskBaselineDuration{

    public static function run($dataDir=null){

        $onesec = 10000000;
        $onemin = 60 * $onesec;
        $onehour = 60 * $onemin;

        # Instantiate project object
        $project = new Project();

        # Creating TaskBaseline:
        $task = $project->getRootTask()->getChildren()->add("Task");
        $baselineType=new BaselineType();
        $project->setBaseline($baselineType->Baseline);

        $baseline = $task->getBaselines()->toList()->get(0);
        $duration = (double)(string)($baseline->getDuration());//.toDouble();
        $baseline_duration = $duration / $onehour;
        print $baseline_duration .PHP_EOL;


    }

}
?>